sdk/keyvault/Azure.Security.KeyVault.Secrets.Extensions.Configuration/src/AzureKeyVaultConfigurationProvider.cs

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+
+namespace Azure.Security.KeyVault.Secrets.Extensions.Configuration
+{
+    /// <summary>
+    /// An AzureKeyVault based <see cref="ConfigurationProvider"/>.
+    /// </summary>
+    internal class AzureKeyVaultConfigurationProvider : ConfigurationProvider, IDisposable
+    {
+        private readonly TimeSpan? _reloadInterval;
+        private readonly SecretClient _client;
+        private readonly IKeyVaultSecretManager _manager;
+        private Dictionary<string, LoadedSecret> _loadedSecrets;
+        private Task _pollingTask;
+        private readonly CancellationTokenSource _cancellationToken;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="AzureKeyVaultConfigurationProvider"/>.
+        /// </summary>
+        /// <param name="client">The <see cref="SecretClient"/> to use for retrieving values.</param>
+        /// <param name="manager">The <see cref="IKeyVaultSecretManager"/> to use in managing values.</param>
+        /// <param name="reloadInterval">The timespan to wait in between each attempt at polling the Azure KeyVault for changes. Default is null which indicates no reloading.</param>
+        public AzureKeyVaultConfigurationProvider(SecretClient client, IKeyVaultSecretManager manager, TimeSpan? reloadInterval = null)
+        {
+            _client = client ?? throw new ArgumentNullException(nameof(client));
+            _manager = manager ?? throw new ArgumentNullException(nameof(manager));
+            if (reloadInterval != null && reloadInterval.Value <= TimeSpan.Zero)
+            {
+                throw new ArgumentOutOfRangeException(nameof(reloadInterval), reloadInterval, nameof(reloadInterval) + " must be positive.");
+            }
+
+            _pollingTask = null;
+            _cancellationToken = new CancellationTokenSource();
+            _reloadInterval = reloadInterval;
+        }
+
+        /// <summary>
+        /// Load secrets into this provider.
+        /// </summary>
+        public override void Load() => LoadAsync().ConfigureAwait(false).GetAwaiter().GetResult();
+
+        private async Task PollForSecretChangesAsync()
+        {
+            while (!_cancellationToken.IsCancellationRequested)
+            {
+                await WaitForReload().ConfigureAwait(false);
+                try
+                {
+                    await LoadAsync().ConfigureAwait(false);
+                }
+                catch (Exception)
+                {
+                    // Ignore
```

Comment:
No logging or tracing at least?

Reply:
`ILogger` is not available in configuration providers and we didn't seem to have a problem with this before. Usually the `Load` method throws first and surfaces the error.